docs/src/ruby/multiple_langs.md

```diff
@@ -33,7 +36,7 @@ Ensure that a `requirements.txt` file, a `package.json` file and a `Gemfile.lock
 $ pack build my-image-name --path .
 :::-- $ touch requirements.txt
 :::-- $ docker rmi -f my-image-name
-:::-> $ pack build my-image-name --path . --pull-policy if-not-present --clear-cache | sed --regexp-extended -e '/Fetching gem metadata/,/Bundle complete/d' -e "/$(date --iso)/d" -e 's/Done \(.+s\)/Done/' -e 's/\.\.\.+ \(.+s\)/.../' -e 's/\b[0-9a-f]{12}\b/abcdef012345/'
```

Comment:
(not blocking) Worth noting that removing this replacement (and some of the others elsewhere) regresses the determinism improvements from #47.

Whilst now having 8 languages means the probability of not having a weekly PR is lower (since at least one of the 8 is likely to have had a change in a week), I wonder if there is still an "ease of reviewing the PR" benefit from the diff churn being less?

Though maybe the weekly PR is already going to be so large that we're pretty much only going to rubber stamp it anyway?

One option to reduce toil might be to make the tutorial PR monthly instead of weekly, though that has downsides too of course. (Perhaps that change is something we do after the CNBs stabilise more?)

Reply:
> (not blocking) Worth noting that removing this replacement (and some of the others elsewhere) regresses the determinism improvements from https://github.com/heroku/buildpacks/pull/47.

Ahh. Those commands fail for bsdsed locally, that's why I took them out:

```
Running: $ '(pack build my-image-name --path . --pull-policy if-not-present --clear-cache | sed --regexp-extended -e '/Fetching gem metadata/,/Bundle complete/d' -e "/$(date --iso)/d" -e 's/Done \(.+s\)/Done/' -e 's/\.\.\.+ \(.+s\)/.../' -e 's/\b[0-9a-f]{12}\b/abcdef012345/') 2>&1' with stdin: '"\n"'
    date: illegal option -- -
    usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
                [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
                [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
    sed: illegal option -- -
    usage: sed script [-Ealnru] [-i extension] [file ...]
    	sed [-Ealnu] [-i extension] [-e script] ... [-f script_file] ... [file ...]
Received exception: #<RuntimeError:"Command `pack build my-image-name --path . --pull-policy if-not-present --clear-cache | sed --regexp-extended -e '/Fetching gem metadata/,/Bundle complete/d' -e \"/$(date --iso)/d\" -e 's/Done \\(.+s\\)/Done/' -e 's/\\.\\.\\.+ \\(.+s\\)/.../' -e 's/\\b[0-9a-f]{12}\\b/abcdef012345/'` exited with non zero status: date: illegal option -- -\nusage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]\n            [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]\n            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]\nsed: illegal option -- -\nusage: sed script [-Ealnru] [-i extension] [file ...]\n\tsed [-Ealnu] [-i extension] [-e script] ... [-f script_file] ... [file ...]\n">, cleaning up before re-raise
## Failed, debug contents are in /Users/rschneeman/Documents/projects/work/buildpacks/docs/fail/ruby
## Moving contents from /var/folders/yr/yytf3z3n3q336f1tj2b2j0gw0000gn/T/d20241217-51449-1ino81 to /Users/rschneeman/Documents/projects/work/buildpacks/docs/fail/ruby
## Cleaning tmp working directory, /var/folders/yr/yytf3z3n3q336f1tj2b2j0gw0000gn/T/d20241217-51449-1ino81
bundler: failed to load command: rundoc (/Users/rschneeman/.gem/ruby/3.3.1/bin/rundoc)
/Users/rschneeman/.gem/ruby/3.3.1/gems/rundoc-4.1.1/lib/rundoc/code_command/bash.rb:53:in `shell': Command `pack build my-image-name --path . --pull-policy if-not-present --clear-cache | sed --regexp-extended -e '/Fetching gem metadata/,/Bundle complete/d' -e "/$(date --iso)/d" -e 's/Done \(.+s\)/Done/' -e 's/\.\.\.+ \(.+s\)/.../' -e 's/\b[0-9a-f]{12}\b/abcdef012345/'` exited with non zero status: date: illegal option -- - (RuntimeError)
usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
            [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
sed: illegal option -- -
usage: sed script [-Ealnru] [-i extension] [file ...]
	sed [-Ealnu] [-i extension] [-e script] ... [-f script_file] ... [file ...]
```

We now have the full power of Ruby we could do something like this ERB where we manually read in and manipulate the contents https://github.com/heroku/buildpacks/pull/63/commits/6f325504647972ee88. That should be consistent across OS-es

Reply:
> One option to reduce toil might be to make the tutorial PR monthly instead of weekly, though that has downsides too of course. (Perhaps that change is something we do after the CNBs stabilise more?)

An (off the wall) alternative to running less: What if we gave ownership of review to product. 🤔 

I think that makes sense. We're in a rapid iteration phase right now. Once things cool off, there should be less churn/updates.